internal/session/session.go

```diff
@@ -355,48 +361,81 @@ func (s *Session) handlePeriodicSearch(ctx context.Context) {
 
 	// TODO: come up with a better strategy for determining when to search
 	// for new providers for blocks.
-	s.sprm.FindMorePeers(ctx, randomWant)
+	s.findMorePeers(ctx, randomWant)
 
 	s.wm.BroadcastWantHaves(ctx, s.id, []cid.Cid{randomWant})
 
 	s.periodicSearchTimer.Reset(s.periodicSearchDelay.NextWaitTime())
 }
 
+// findMorePeers attempts to find more peers for a session by searching for
```

Comment:
I brought this function from inside of the SessionPeerManager to be in the Session instead, as it makes it easier to follow whats going on and simplifies the dependencies between Session, sessionWantSender and SessionPeerManager